src/Native/Runtime/arm/ExceptionHandling.S

```diff
@@ -56,23 +86,70 @@ NESTED_END RhpThrowHwEx
 //
 // RhpThrowEx
 //
-// INPUT:  RDI:  exception object
+// INPUT:  R0:  exception object
 //
 // OUTPUT:
 //
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////
 NESTED_ENTRY RhpThrowEx, _TEXT, NoHandler
 
-ALTERNATE_ENTRY RhpThrowEx2
-
-#ifdef _DEBUG
-        GEN_ASSERT_FUNC RhpThrowExName
-#endif
+        // Setup a PAL_LIMITED_CONTEXT on the stack {
+        vpush       {d8-d15}
+        PROLOG_PUSH "{r0,lr}"        	// Reserve space for SP and store LR
+        PROLOG_PUSH "{r0,r4-r11,lr}"
+        // } end PAL_LIMITED_CONTEXT
+
+        PROLOG_STACK_ALLOC STACKSIZEOF_ExInfo
+
+        mov         r4, r0 // Save exception object
+        // r0 = GetThread()
+        INLINE_GETTHREAD
+
+        // r2 <- addr of return address
+        add         r2, sp, #(rsp_offsetof_Context + SIZEOF__PAL_LIMITED_CONTEXT + 0x8)
+
+        // There is runtime C# code that can tail call to RhpThrowEx using a binder intrinsic. So
+        // the return address could have been hijacked when we were in that C# code and we must
+        // remove the hijack and reflect the correct return address in our exception context re-
+        // cord. The other throw helpers don't need this because they cannot be tail-called from C#.
+        INLINE_THREAD_UNHIJACK  r0, r1, r3        // trashes R1, R3
```

Comment:
The Windows version is different - look for `NOTE: we cannot use INLINE_THREAD_UNHIJACK ...`. It would be nice for the Unix version to be in sync.